shell/wayland/display.cc

```diff
@@ -803,3 +826,45 @@ const struct wl_surface_listener Display::base_surface_listener = {
     .enter = handle_base_surface_enter,
     .leave = handle_base_surface_leave,
 };
+
+void Display::agl_shell_bound_ok(void* data, struct agl_shell* shell) {
+  (void)shell;
+  auto* d = static_cast<Display*>(data);
+  d->m_agl.wait_for_bound = false;
+  d->m_agl.bound_ok = true;
+}
+
+void Display::agl_shell_bound_fail(void* data, struct agl_shell* shell) {
+  (void)shell;
+  auto* d = static_cast<Display*>(data);
+  d->m_agl.wait_for_bound = false;
+  d->m_agl.bound_ok = false;
+}
+
+void Display::agl_shell_app_state(void* data,
+                                  struct agl_shell* agl_shell,
+                                  const char* app_id,
+                                  uint32_t state) {
+  (void)data;
+  (void)agl_shell;
+
+  switch (state) {
+    case AGL_SHELL_APP_STATE_STARTED:
+      FML_DLOG(INFO) << "[" << app_id << "] AGL_SHELL_APP_STATE_STARTED";
```

Comment:
Here you'll need to call activate_app() request. 

If you read out my longer explanation of why we need v3 is because indirectly the compositor will map/activate/display the application at start-up, but I'd like to re-work that to be driven by the shell client, rather than indirectly by the compositor, so here you'll need to have that in.

Note that until you explicitly edit the configuration file and under the core section add activate-by-default=false this won't really take effect. 

See also the alternative I've been looking into as to allow for a transition to work out: https://gerrit.automotivelinux.org/gerrit/c/AGL/meta-agl-demo/+/27991